multiversx/mx-chain-go
# Dynamic gas data trie loads

## Reasoning behind the pull request
- The new feature of dynamic gas cost computation based on trie loads needs to be activated at a certain epoch by all nodes.
  
## Proposed changes
- Add a new flag `DynamicGasCostForDataTrieStorageLoadEnableEpoch` that will enable at the specified epoch the computation of gas cost for data trie loads based on the depth of the trie.

## Testing procedure
- Create a smart contract with multiple leaves in the dataTrie. When a leaf is accessed, check that the gas cost is computed dynamically based on the depth of the leaf.

## Pre-requisites

Based on the [Contributing Guidelines](https://github.com/ElrondNetwork/elrond-go/blob/master/.github/CONTRIBUTING.md#branches-management) the PR author and the reviewers must check the following requirements are met:
- was the PR targeted to the correct branch?
- if this is a larger feature that probably needs more than one PR, is there a `feat` branch created?
- if this is a `feat` branch merging, do all satellite projects have a proper tag inside `go.mod`?
